### PR TITLE
Report sftp_error when write fails

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

Add additional error reporting to SFTP write errors

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

N/A

### CODE REVIEW

**Comment by webknjaz** [comment]
@Qalthos this needs a changelog fragment and maybe a test? Oh and please fill out the summary.

**Review by ganeshrn — APPROVED** [review]

**Comment by webknjaz** [comment]
@Qalthos is https://github.com/ansible/pylibssh/issues/222 related?

**Comment by webknjaz** [comment]
@Qalthos plz rebase to pull in the CI+linter+docs build fixes from devel.

File: docs/changelog-fragments/216.bugfix.2.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Changed ``sftp.sftp_get`` to write files as bytes rather than assuming files are valid UTF8 -- by :user: `Qalthos`
```
Comment (webknjaz) [inline docs/changelog-fragments/216.bugfix.2.rst]: @Qalthos FYI this is a syntax error in RST :point_up: — there shouldn't be a space between the role `:user:` and its argument for Sphinx to parse that as one thing. Otherwise, it tries to interpret them separately.

This caused the failures here:
* https://github.com/ansible/pylibssh/pull/216/checks?check_run_id=2806860150
* https://readthedocs.org/projects/ansible-pylibssh/builds/13990866/

```suggestion
Changed ``sftp.sftp_get`` to write files as bytes rather than assuming files are valid UTF8 -- by :user:`Qalthos`
```

File: docs/changelog-fragments/216.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Added additional error reporting to SFTP write errors -- by :user: `Qalthos`
```
Comment (webknjaz) [inline docs/changelog-fragments/216.bugfix.rst]:
```suggestion
Added additional error reporting to SFTP write errors -- by :user:`Qalthos`
```

File: docs/changelog-fragments/216.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Added additional error reporting to SFTP write errors -- by :user:`Qalthos`
```
Comment (webknjaz) [inline docs/changelog-fragments/216.bugfix.rst]: Is this one really a bugfix? Do you want to use a fragment number here too, to control ordering in the changelog?

Comment (webknjaz) [reply docs/changelog-fragments/216.bugfix.rst]: The order is visible here: https://ansible-pylibssh--216.org.readthedocs.build/en/216/changelog/#vrelease-unreleased-draft-2021-06-16

**Comment by webknjaz** [comment]
@Qalthos this needs rebasing

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -57,7 +57,7 @@ repos:
       ^docs/_samples/.*\.py$
 
 - repo: https://gitlab.com/pycqa/flake8.git
-  rev: 3.8.2
+  rev: 3.9.2
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: Why did you change it in this PR?

File: bin/pep517_backend/_backend.py [hunk 5]
```diff
@@ -76,15 +76,12 @@ def get_config():
 
         [tool.local.cythonize.kwargs.directives]
         # This section can contain compiler directives
-        # NAME = "VALUE"
```
Comment (webknjaz) [inline bin/pep517_backend/_backend.py]: This should remain as it's the only example of how things are supposed to be configured right now. You could either uncomment it or extend the exclusion at https://github.com/ansible/pylibssh/blob/820a2d0/.flake8#L86.

File: .pre-commit-config.yaml (hunk 4 above)
Comment (Qalthos) [reply .pre-commit-config.yaml]: Oh you're right, I was trying things out and did not undo that

File: bin/pep517_backend/_backend.py (hunk 5 above)
Comment (Qalthos) [reply bin/pep517_backend/_backend.py]: Ah, so that is how that is configured. That seems much more sensible.

Comment (webknjaz) [reply bin/pep517_backend/_backend.py]: Yeah, I also think that it's a false-positive because it's a comment inside a docstring. Probably need to report this to `eradicate`.

Comment (webknjaz) [reply bin/pep517_backend/_backend.py]: Looks like it's https://github.com/wemake-services/flake8-eradicate/issues/215.

**Comment by webknjaz** [comment]
FYI don't worry about the Fedora:34 job, it's not related to this PR.

File: .flake8 [hunk 6]
```diff
@@ -83,7 +83,7 @@ pytest-parametrize-values-row-type = tuple
 
 # flake8-eradicate
 # E800:
-eradicate-whitelist-extend = distutils:\s+libraries\s+=\s+|isort:\s+\w+
+eradicate-whitelist-extend = distutils:\s+libraries\s+=\s+|isort:\s+\w+|NAME = "VALUE"
```
Comment (webknjaz) [inline .flake8]: Maybe you could submit this separately with its own misc change note?

Comment (Qalthos) [reply .flake8]: Yeah, I'm doing that now
